src/test/java/net/minecraftforge/test/DependencyExtractionTest.java

```diff
@@ -0,0 +1,117 @@
+package net.minecraftforge.test;
+
+import net.minecraftforge.fml.relauncher.DependencyExtractor;
+import net.minecraftforge.fml.relauncher.DependencyExtractor.Artifact;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.IOUtils;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.jar.JarFile;
+
+import static org.junit.Assert.*;
+
+public class DependencyExtractionTest
+{
+    /**
+     * Base 64 encoded JAR file of the following structure:
+     * container.jar
+     * ├─── META-INF
+     * │   └    MANIFEST.MF (contains 'ContainedDeps: dependency.jar' definition)
+     * └─── dependency.jar
+     *     └─── META-INF
+     *         └    MANIFEST.MF (contains 'Maven-Artifact: net.minecraftforge.test:dependency:1.0:testing' definition)
+     */
+    private static final String JAR_BASE64 =
```

Comment:
This would probably be more clear if it was just included as a jar file alongside the test.